types/wlr_screencopy_v1.c

```diff
@@ -8,10 +8,85 @@
 #include "wlr-screencopy-unstable-v1-protocol.h"
 #include "util/signal.h"
 
-#define SCREENCOPY_MANAGER_VERSION 1
+#define SCREENCOPY_MANAGER_VERSION 2
+
+struct screencopy_damage {
+	struct wl_list link;
+	struct wlr_output *output;
+	struct pixman_region32 damage;
+	struct wl_listener output_precommit;
```

Comment:
We need an output destroy listener here as well, in case the output is removed while recording.